class.jetpack-cli.php

```diff
@@ -1476,6 +1476,156 @@ private function partner_provision_error( $error ) {
+		}
+
+		global $wp_filesystem;
+		if ( ! WP_Filesystem() ) {
+			WP_CLI::error( esc_html__( "Can't write files", 'jetpack' ) . ' 😱' );
+		}
+
+		$path = JETPACK__PLUGIN_DIR . "extensions/blocks/$slug";
+
+		if ( $wp_filesystem->exists( $path ) && $wp_filesystem->is_dir( $path ) ) {
+			WP_CLI::error( sprintf( esc_html__( 'Name conflicts with the existing block %s', 'jetpack' ), $path ) . ' ⛔️' );
+			exit( 1 );
+		}
+
+		$wp_filesystem->mkdir( $path );
+
+		$hasKeywords = isset( $assoc_args['keywords'] );
+
+		$files = array(
+			"$path/$slug.php" => $this->render_block_file( 'block-register-php', array(
+				'slug' => $slug,
+				'title' => $title,
+				'underscoredSlug' => str_replace( '-', '_', $slug ),
+			) ),
+			"$path/index.js" => $this->render_block_file( 'block-index-js', array(
+				'slug' => $slug,
+				'title' => $title,
+				'description' => isset( $assoc_args['description'] )
+					? $assoc_args['description']
+					: $title,
+				'keywords' => $hasKeywords
+					? array_map( function( $keyword ) {
+						// Construction necessary for Mustache lists
+						return array( 'keyword' => trim( $keyword ) );
+					}, explode( ',', $assoc_args['keywords'] ) )
+					: '',
+				'hasKeywords' => $hasKeywords
+			) ),
+			"$path/editor.js" => $this->render_block_file( 'block-editor-js' ),
+			"$path/editor.scss" => $this->render_block_file( 'block-editor-scss', array(
+				'slug' => $slug,
+				'title' => $title,
+			) ),
+			"$path/edit.js" => $this->render_block_file( 'block-edit-js', array(
+				'title' => $title,
+				'className' => str_replace( ' ', '', ucwords( str_replace( '-', ' ', $slug ) ) ),
+			) )
+		);
+
+		$files_written = array();
+
+		foreach ( $files as $filename => $contents ) {
+			if ( $wp_filesystem->put_contents( $filename, $contents ) ) {
+				$files_written[] = $filename;
+			} else {
+				WP_CLI::error( sprintf( esc_html__( 'Error creating %s', 'jetpack' ), $filename ) );
+			}
+		}
+
+		if ( empty( $files_written ) ) {
+			WP_CLI::log( esc_html__( 'No files were created' ) );
+		} else {
+			// Load index.json and insert the slug of the new block in the production array
+			$block_list_path = JETPACK__PLUGIN_DIR . 'extensions/index.json';
+			$block_list = $wp_filesystem->get_contents( $block_list_path );
+			if ( empty( $block_list ) ) {
+				WP_CLI::error( sprintf( esc_html__( 'Error fetching contents of %s', 'jetpack' ), $block_list_path ) );
+			} else if ( false === stripos( $block_list, $slug ) ) {
+				$new_block_list = json_decode( $block_list );
+				$new_block_list->beta[] = $slug;
+				if ( ! $wp_filesystem->put_contents( $block_list_path, wp_json_encode( $new_block_list ) ) ) {
+					WP_CLI::error( sprintf( esc_html__( 'Error writing new %s', 'jetpack' ), $block_list_path ) );
+				}
+			}
+
+			WP_CLI::success( sprintf( esc_html__( 'Successfully created block %s with slug %s', 'jetpack' ) . ' 🎉', $title, $slug ) );
```

Comment:
What do you think about adding more details about potential next steps here? Maybe a link like https://github.com/Automattic/jetpack/blob/master/extensions/README.md#develop-new-blocks for example? It may also be worth mentioning that the block was added as a Beta block, and let folks know they need the constant to see the block they just created.

Reply:
Done, thanks

Reply:
Thanks! I was thinking we could add something along the lines of "Done! Only one last step see your new block in the editor: run `yarn build-extensions`", or something like that.

Reply:
I would try to keep references to exact commands to run in fewer places (i.e. docs) and refer folks to read them, so that there are less places to maintain. What do you think?

Mentioning beta constant would make sense here, so that folks who skip reading docs and think they know what they're doing won't get lost. :-)

Reply:
True, that makes sense. Can we say something like "you have to build the blocks" though? I feel like that's something that may not be obvious when you first run the command / start working on blocks in Jetpack.

Reply:
Yup, something like that would be good, 👍

Reply:
I like it 👍

Reply:
😍